server/src/main/java/org/elasticsearch/index/snapshots/blobstore/SlicedInputStream.java

```diff
@@ -83,14 +92,17 @@ public final int read(byte[] buffer, int offset, int length) throws IOException
             nextStream();
             return read(buffer, offset, length);
         }
+        currentSliceOffset += read;
         return read;
     }
 
     @Override
-    public final void close() throws IOException {
+    public void close() throws IOException {
         IOUtils.close(currentStream);
+        closed = true;
```

Comment:
I think we should do this prior to closing `currentStream`, just in case that throws.

Reply:
I moved the closing of the currentStream last.